--- class_password-methods.inc	2019-05-29 16:21:55.605320688 +0200
+++ class_password-methods.inc	2019-05-29 16:22:42.409712203 +0200
@@ -180,9 +180,11 @@
       if ($mode == 'LOCK') {
         /* Lock entry */
         $pwd = preg_replace("/(^[^\}]+\})(.*$)/",   "\\1!\\2",  $pwd);
+        $modify['description'] = 'locked';
       } else {
         /* Unlock entry */
         $pwd = preg_replace("/(^[^\}]+\})!(.*$)/",  "\\1\\2",   $pwd);
+        $modify['description'] = 'unlocked';
       }
       $modify['userPassword'] = $pwd;
       $ldap->cd($dn);
