--- class_password-methods-crypt.inc	2020-03-05 16:59:00.396418933 +0100
+++ class_password-methods-crypt.inc	2020-03-05 17:07:30.610602507 +0100
@@ -153,6 +153,10 @@
    */
   static function _extract_method($classname, $password_hash)
   {
+    if (preg_match('/^{SHA256}/i', $password_hash)) {
+      return "crypt/sha-256";
+    }
+
     if (!preg_match('/^{crypt}/i', $password_hash)) {
       return "";
     }
